tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/CiamIntegrationTests.cs

```diff
@@ -144,6 +149,74 @@ public async Task ClientCredentialWithClientSecret_Ciam_Async(string authorityFo
             Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
         }
 
+        [TestMethod]
+        public async Task Ciam_Cud_Obo_Test_Async()
+        {
+            string authorityNonCud = "https://MSIDLABCIAM6.ciamlogin.com";
+            string authorityCud = "https://login.msidlabsciam.com/fe362aec-5d43-45d1-b730-9755e60dc3b9/v2.0/";
+            string ciamWebapp = "b244c86f-ed88-45bf-abda-6b37aa482c79";
+            string ciamWebApi = "634de702-3173-4a71-b336-a4fab786a479";
+            string ciamEmail = "idlab@msidlabciam6.onmicrosoft.com";
+
+            //Get lab details
+            var labResponse = await LabUserHelper.GetLabUserDataAsync(new UserQuery()
+            {
+                FederationProvider = FederationProvider.CIAM,
+                SignInAudience = SignInAudience.AzureAdMyOrg,
+                PublicClient = PublicClient.no
+            }).ConfigureAwait(false);
+
+            //Acquire tokens
+            var msalPublicClient = PublicClientApplicationBuilder
+                .Create(ciamWebapp)
+                .WithAuthority(authorityNonCud, false)
+                .WithRedirectUri(_ciamRedirectUri)
+                .Build();
+
+            var result = await msalPublicClient
+                .AcquireTokenByUsernamePassword(new[] { $"api://{ciamWebApi}/.default" }, ciamEmail, LabUserHelper.FetchUserPassword("msidlabciam6"))
```

Comment:
Why is this `LabUserHelper.FetchUserPassword` a public API? I thought we have a "labResponse.User.GetPassword" API already. Generally it's not a good idea to hvae 2 APIs that do the same thing.

Reply:
There are some rare cases where the user does not exist in the lab api so this can be useful. Not sure why it is public though, this code is kinda old and there have been a lot of iterations.